+sw_tests/+unit_tests/unittest_spinw.m

```diff
@@ -0,0 +1,119 @@
+classdef unittest_spinw < matlab.mock.TestCase
+    % Runs through unit tests for @spinw/spinw.m
+    properties (TestParameter)
+        spinw_struct_input = { ...
+            {struct('lattice', struct('angle', [pi, pi, (2*pi)/3], ...
+                                     'lat_const', [2, 2, 4])),
+             'spinw_from_struct_lat224.mat'}, ...
+            {struct('lattice', struct('angle', [pi; pi; (2*pi)/3], ...
+                                     'lat_const', [2; 2; 4])), ...
+             'spinw_from_struct_lat224.mat'}};
+        spinw_figure_input = { ...
+            {'default_structure.fig', 'spinw_default.mat'}, ...
+            {'afm_chain_spec.fig', 'spinw_afm_chain.mat'}}
+        spinw_file_input = { ...
+            {'YFeO3_mcphase.cif', 'spinw_from_cif_YFeO3_mcphase.mat'}, ...
+            {'LaFeO3_fullprof.cif', 'spinw_from_cif_LaFeO3_fullprof.mat'}, ...
+            {'BiMn2O5.fst', 'spinw_from_fst_BiMn2O5.mat'}};
+    end
+    methods (Static)
+        function udir = get_unit_test_dir()
+            udir = fullfile('.', 'test_data', 'unit_tests');
+        end
+    end
+    methods
+        function obj = load_spinw(testCase, filename)
+            obj = load(fullfile(testCase.get_unit_test_dir(), 'spinw', filename));
+            obj = obj.data;
+        end
+        function obj = load_figure(testCase, filename)
+            path = fullfile(testCase.get_unit_test_dir(), 'Figure', filename);
+            obj = openfig(path, 'invisible');
+        end
+        function verify_obj(testCase, expected_obj, actual_obj, rel_tol, abs_tol)
+            if nargin < 5
+                abs_tol = 0;
+            end
+            if nargin < 4
+                rel_tol = 1e-10;
+            end
+            all_fieldnames = union(fieldnames(expected_obj), fieldnames(actual_obj));
```

Comment:
Here should we not add an additional assertion that  f `expected_obj` and `actual_obj` have different numbers of fieldnames they are not equal? We can also assert that the objects are of the same class type?

Reply:
Yeah good idea on testing the class type! And I wanted to test the combined fieldnames of both objects, if we just use the fields from the expected object, if the actual object has extra fields that have been added accidentally this wont be detected and the test will still pass.